rtos/Thread.h

```diff
@@ -540,7 +540,7 @@ class Thread : private mbed::NonCopyable<Thread> {
     bool                       _dynamic_stack;
     Semaphore                  _join_sem;
     mutable Mutex              _mutex;
-    mbed_rtos_storage_thread_t _obj_mem;
+    mbed_rtos_storage_thread_t _obj_mem = {};
```

Comment:
The current init is in start() method - it sets this to 0: https://github.com/ARMmbed/mbed-os/blob/b1ced1bb5b0b3df13ee72ba71fb485470885eca1/rtos/source/Thread.cpp#L109 

Assuming based on how it is initialized (POD type), this could be handled in the constructor as the rest of members - to be consistent however this is valid now as it is.

Reply:
I'm not a fan of the existing `constructor` method and its in-body init - now we're C++11 you could use proper constructor delegation, and the final most-flexible constructor could do everything via its member-initializer list.

I'd also prefer if everything was consistent - all defaults in body of `constructor`, all defaults in member-initializer-list, or all defaults as default-member initializers.

So make it a `memset` in constructor, or do a more wide-ranging refactor/tidy including real constructor delegation.